quickshell-git fails to build on Apple Silicon (Asahi): CMake can't find OpenGL/WrapOpenGL for Qt6Gui

## Summary

`omarchy-upgrade-to-quattro-mac` fails to build `quickshell-git` on Apple Silicon (Asahi Linux). The AUR build fails at CMake configuration because `find_package(OpenGL)` can't satisfy Qt6Gui's `WrapOpenGL` requirement, even though `mesa`, `libglvnd`, `libGL.so`, and the GL headers are all installed. This blocks the 3.x → Quattro upgrade entirely, since quickshell is the core Quickshell-based shell Quattro switches to.

## Environment

- Hardware: Apple MacBook Pro (16-inch, M2 Max, 2023)
- Kernel: `7.1.6-1-1-ARCH aarch64` (Asahi/Arch Linux ARM)
- `qt6-base` 6.11.2-2
- `mesa` 26.1.7-1
- `libglvnd` 1.7.0-3
- `cmake` 4.4.2-1
- Package attempted: `quickshell-git 0.3.1.r1.g0fed22a-1` (AUR)
- Omarchy Mac was on v3.8.2, upgrading via `omarchy-upgrade-to-quattro-mac --yes --ref quattro`

## Steps to reproduce

1. On an Apple Silicon Mac running Omarchy Mac 3.x (Asahi Linux)
2. Run `curl -fsSL https://raw.githubusercontent.com/omarchy-mac/omarchy-mac/quattro/bin/omarchy-upgrade-to-quattro-mac | bash`
3. Wait for the package install phase to reach `quickshell-git`

## Actual behavior

The `quickshell-git` AUR build fails during CMake configuration:

```
-- Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY)
-- Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OpenGL)
-- Could NOT find WrapOpenGL (missing: WrapOpenGL_FOUND)
CMake Warning at /usr/lib/cmake/Qt6/Qt6Config.cmake:267 (find_package):
  Found package configuration file:
    /usr/lib/cmake/Qt6Gui/Qt6GuiConfig.cmake
  but it set Qt6Gui_FOUND to FALSE so package "Qt6Gui" is considered to be
  NOT FOUND.  Reason given by package:
  Qt6Gui could not be found because dependency WrapOpenGL could not be found.

CMake Error at CMakeLists.txt:141 (find_package):
  Found package configuration file:
    /usr/lib/cmake/Qt6/Qt6Config.cmake
  but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
  FOUND.  Reason given by package:
  Failed to find required Qt component "Gui".

-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
 -> error making: quickshell-git-exit status 4
```

`omarchy-upgrade-to-quattro-mac`'s `install_quattro_packages` step catches this and continues (logging it under "Skipped packages that would not install on aarch64"), so the upgrade proceeds without quickshell installed. Later, the `1784672586` migration ("Switch to the Omarchy quickshell-git build...") then fails outright with `error: target not found: quickshell-git`, because it calls `sudo pacman -S quickshell-git` directly — `pacman` can't resolve AUR packages, so this fails even when `quickshell-git` *is* buildable, on top of the underlying build failure.

## Suspected root cause

`OPENGL_glx_LIBRARY` is missing — CMake's standard `FindOpenGL` module defaults to looking for desktop GLX (X11) unless told otherwise. This is a headless/Wayland-only Asahi system with no GLX vendor registered via glvnd (no X server, and the Asahi GPU driver stack may only register EGL, not GLX). `mesa`/`libglvnd` are present and provide `libGL.so`/`GL/gl.h`, but that alone doesn't satisfy CMake's GLX check. Quickshell's `CMakeLists.txt` (or Qt6Gui's own `find_package(OpenGL)` call under the hood) doesn't appear to account for this — possibly needs `-DOPENGL_egl_LIBRARY`/EGL-only detection, or a PKGBUILD patch specific to Wayland-only ARM systems.

## Other findings from the same upgrade attempt (may be worth separate issues)

The same run's package pass also reported these as hard-incompatible with aarch64 (their AUR/PKGBUILD only support x86_64): `asdcontrol`, `obs-studio`, `omacalc`, `omacut`, `omawrite`, `omarchy-nvim`, `pinta`, `qemu-user-static-binfmt`, `tobi-try`, `ttf-jetbrains-mono-nerd-basic`. (`omarchy-nvim` overlaps with the already-tracked #203 "Add neovim package for aarch64".)

## Expected behavior

`quickshell-git` should build successfully on Apple Silicon/Asahi so the Quattro upgrade can complete and provide a working desktop shell.

--- a/bin/omarchy-upgrade-to-quattro-mac
+++ b/bin/omarchy-upgrade-to-quattro-mac
@@ -134,6 +134,26 @@ switch_checkout_to_quattro() {
   git -C "$checkout" merge --ff-only "origin/$upgrade_ref"
 }
 
+# A 3.x machine's /etc/pacman.conf predates the [omarchy-aarch64] repo. Without
+# it, every Quattro package that the ARM repo serves as a prebuilt binary
+# resolves to an hours-long AUR build instead -- quickshell-git's build even
+# fails outright on Wayland-only systems whose GL stack predates the upgrade
+# (#208). Mirrors install.sh's ensure_arm_package_repo; must run before
+# install_quattro_packages.
+ensure_arm_package_repo() {
+  grep -q '^\[omarchy-aarch64\]' /etc/pacman.conf && return 0
+
+  local block
+  block=$(sed -n '/^\[omarchy-aarch64\]/,/^Server[[:space:]]*=/p' \
+    "$checkout/default/pacman/pacman-stable.conf")
+  [[ -n $block ]] || fail "default/pacman/pacman-stable.conf has no [omarchy-aarch64] section."
+
+  log "Adding the Omarchy ARM package repo"
+  printf '\n%s\n' "$block" | sudo tee -a /etc/pacman.conf >/dev/null
+  sudo pacman -Sy --noconfirm >/dev/null 2>&1 ||
+    warn "Could not refresh package databases after adding the ARM repo."
+}
+
 install_quattro_packages() {
   local package skipped=()
 
@@ -276,6 +296,7 @@ switch_to_networkmanager() {
 main() {
   backup_user_config
   switch_checkout_to_quattro
+  ensure_arm_package_repo
   install_quattro_packages
   wire_system_paths
   update_bashrc

--- a/install/omarchy-aarch64-unavailable.packages
+++ b/install/omarchy-aarch64-unavailable.packages
@@ -24,14 +24,13 @@ obs-studio
 # emitting a misleading "skipped" warning for a package that cannot install here.
 dotnet-runtime
 
-# quickshell-git builds on aarch64 -- and then the build is thrown away. The
-# omarchy package depends on plain `quickshell`, which Arch Linux ARM now ships
-# in extra (0.3.0-2), so pacman satisfies the dependency from the repo and the
-# freshly compiled quickshell-git is never installed. Measured on a clean
-# install: the artifact sat unused in yay's cache while `pacman -Qi quickshell`
-# reported the repo package installed as a dependency of omarchy. Skipping it
-# changes nothing about what ends up on the machine, and drops a Qt/C++ compile
-# plus its cli11 and vulkan-headers makedepends from every install.
+# quickshell-git builds on aarch64, but the package pass skips it anyway: the
+# omarchy package depends on plain `quickshell`, which Arch Linux ARM ships in
+# extra (0.3.0-2), and installing both in one pass would fight over the
+# conflict. Migration 1784672586 swaps plain quickshell for the quickshell-git
+# binary from the [omarchy-aarch64] repo at login; until that binary is
+# published, the desktop stays on the repo package and only synchronous shell
+# restarts wait-for-exit is lost.
 quickshell-git
 
 # pinta is arch=('x86_64') in the AUR, so makepkg refuses it in seconds -- but

--- a/migrations/1784672586.sh
+++ b/migrations/1784672586.sh
@@ -1,6 +1,17 @@
 echo "Switch to the Omarchy quickshell-git build so shell restarts wait for instance exit"
 
 if ! omarchy-pkg-present quickshell-git; then
+  # pacman only resolves packages from sync repos, and quickshell-git is served
+  # by the [omarchy-aarch64] repo. If this machine cannot see it there -- a 3.x
+  # pacman.conf predating the repo section, or the binary not published yet --
+  # skip instead of failing: a failed migration aborts omarchy-migrate's whole
+  # run and retries at every login. Plain quickshell keeps the desktop working;
+  # only synchronous shell restarts wait-for-exit is lost until it appears.
+  if ! pacman -Si quickshell-git &>/dev/null; then
+    echo "quickshell-git is not in any configured repo; keeping plain quickshell" >&2
+    exit 0
+  fi
+
   # One transaction with --ask 4 so pacman accepts replacing the conflicting
   # quickshell package in place; packages depending on quickshell stay
   # satisfied through the provides.

--- a/test/shell.d/aarch64-packages-test.sh
+++ b/test/shell.d/aarch64-packages-test.sh
@@ -60,3 +60,27 @@ set_call=$(grep -n '^  install_default_package_set$' "$ROOT/install.sh" | cut -d
 (( repo_call < set_call )) ||
   fail "the ARM repo is added before the default package set is installed"
 pass "the ARM repo is added before the default package set is installed"
+
+# The Quattro upgrade has the same trap with a twist: a 3.x machine's
+# /etc/pacman.conf predates the ARM repo entirely, and installing packages
+# first sent quickshell-git to an AUR build that fails on Wayland-only GL
+# stacks (#208). The upgrade must wire the repo in from its own fresh checkout
+# before the package pass.
+repo_call=$(grep -n '^  ensure_arm_package_repo$' "$ROOT/bin/omarchy-upgrade-to-quattro-mac" | cut -d: -f1)
+checkout_call=$(grep -n '^  switch_checkout_to_quattro$' "$ROOT/bin/omarchy-upgrade-to-quattro-mac" | cut -d: -f1)
+set_call=$(grep -n '^  install_quattro_packages$' "$ROOT/bin/omarchy-upgrade-to-quattro-mac" | cut -d: -f1)
+[[ -n $repo_call && -n $checkout_call && -n $set_call ]] ||
+  fail "the upgrade switches checkout, adds the ARM repo, and installs the set"
+(( checkout_call < repo_call )) ||
+  fail "the ARM repo block is read from the Quattro checkout, so switch first"
+(( repo_call < set_call )) ||
+  fail "the upgrade adds the ARM repo before installing the Quattro set"
+pass "the Quattro upgrade adds the ARM repo before installing packages"
+
+# Migration 1784672586 swaps plain quickshell for quickshell-git with a raw
+# pacman call, which only resolves sync-repo targets. Without an availability
+# guard, machines that cannot see the ARM repo fail the migration outright --
+# and one failed migration aborts omarchy-migrate's whole run at every login.
+grep -qF 'pacman -Si quickshell-git' "$ROOT/migrations/1784672586.sh" ||
+  fail "the quickshell-git migration checks sync-repo availability before pacman"
+pass "the quickshell-git migration skips cleanly when the repo cannot serve it"

